src/services/transfers.service.ts

```diff
@@ -10,20 +10,26 @@ export type TransfersByCursorOptions = {
 
 class TransfersService {
   public transfers = new PrismaClient().transfer
-  private currentCursor: string | undefined
 
-  private prepareQueryParams(args: TransfersByCursorOptions): { pageSize: number; skip: number; where: (TransferStatus & { sender: string }) | {} } {
+  private prepareQueryParams(args: TransfersByCursorOptions): {
+    pageSize: number
+    skip: number
+    take: number
+    where: (TransferStatus & { sender: string }) | {}
+  } {
     const { page, limit, ...rest } = args
 
     const pageSize = parseInt(limit, 10)
-    const pageIndex = parseInt(page, 10) - 1
-    const skip = pageIndex * pageSize
+    const pageIndex = parseInt(page, 10)
+    const skip = (pageIndex - 1) * pageSize
+    const take = pageSize
```

Comment:
nothing special, I will remove pageSize, it was being used to slice the final array when the solution was returning, after the query, the elements from the starting point times the page size:

ie: page 2, limit 10 => take was 20 and skip 10. Transfers returned: from 20 to 39 records. Then since page size was 10, the `transfer.slice(0, pageSize)` give back the 10 transfers needed. I forgot to remove that parameter.